core/src/test/scala/com/rackspace/com/papi/components/checker/wadl/BaseCheckerSpec.scala

```diff
@@ -428,28 +429,28 @@ class BaseCheckerSpec extends BaseWADLSpec {
 
     if (step_funs.isEmpty) throw new TestFailedException("Path assertion should contain at least one step!",4)
 
-    def followPath(step : NodeSeq, nextSteps : Seq[NodeSeq]) : Boolean = {
-      if (nextSteps.isEmpty) return true
-      val next = nextSteps(0)
-      step.foreach(n =>
-        (n \ "@next").text.split(" ").foreach(m => next.foreach (o =>
-          if ((o \ "@id").text == m) {
-            if (followPath (next, nextSteps.drop(1))) return true
-          })))
-      false
-    }
 
-    val steps = step_funs.map (n => n(checker))
+    def getNext(step : Node) : NodeSeq = {
+      (step \ "@next").text.split(" ").map(id => { (checker \\ "step").filter (ns => { (ns \ "@id").text == id}) }).fold(NodeSeq.Empty)(_ ++ _)
+    }
 
-    //
-    //  Check to make sure all the steps are available...
-    //
-    steps.foreach (s => if (s.isEmpty) throw new TestFailedException("A step in the path could not be found", 4))
+    def followPath(steps : NodeSeq, stepFuns : List[(NodeSeq) => NodeSeq]) : Boolean = stepFuns match {
+      case Nil => true
+      case stepFun :: nextFuns =>
+        val matchSteps = stepFun(steps)
+        logger.trace ("Match Steps "+matchSteps)
+        if (matchSteps.isEmpty) return false
```

Comment:
The full quote is:

_(emphasis mine)_
>The recommended style for methods is in fact to avoid having explicit, and especially multiple, return statements. Instead, think of each method as an expression that yields one value, which is returned. This philosophy will encourage you to make methods quite small, to factor larger methods into multiple smaller ones. **On the other hand, design choices depend on the design context, and Scala makes it easy to write methods that have multiple, explicit returns if that's what you desire.**    


That being said, is that line even necessary?  I imagine the `map/fold(false)` would manage the case of `matchSteps` being empty and will return `false` on its own without much overhead (since the list is empty).